pkg/operations/deletevm.go

```diff
@@ -86,5 +91,25 @@ func CleanDeleteVirtualMachine(az armhelpers.ACSEngineClient, logger *log.Entry,
 		}
 	}
 
+	// Role assignments are not deleted if the VM is destroyed, so we must cleanup ourselves!
+	// The role assignments should only be relevant if managed identities are used,
+	// but always cleaning them up is easier than adding rule based logic here and there.
+	scope := fmt.Sprintf(AADRoleResourceGroupScopeTemplate, subscriptionID, resourceGroup)
+	logger.Infof("fetching roleAssignments: %s with principal %s", scope, *vm.Identity.PrincipalID)
```

Comment:
Ran an upgrade test against this branch and got the following nil pointer panic:

```
time="2018-05-15T19:03:56Z" level=info msg="deleting managed disk: kubernetes-koreasouth-74251/k8s-master-15172440-0_OsDisk_1_dc7dab13d2354a4f875595aa83b99d3c"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x138a30f]

goroutine 1 [running]:
github.com/Azure/acs-engine/pkg/operations.CleanDeleteVirtualMachine(0x186eca0, 0xc4205f4900, 0xc42031c780, 0xc4208ade60, 0x24, 0x7fff98924da7, 0x1b, 0xc42034bc60, 0x15, 0x148c120, ...)
	/go/src/github.com/Azure/acs-engine/pkg/operations/deletevm.go:98 +0xa2f
github.com/Azure/acs-engine/pkg/operations/kubernetesupgrade.(*UpgradeMasterNode).DeleteNode(0xc4201622a0, 0xc42086eb20, 0xc42034bc00, 0x15, 0x0)
	/go/src/github.com/Azure/acs-engine/pkg/operations/kubernetesupgrade/upgrademasternode.go:39 +0x7f
github.com/Azure/acs-engine/pkg/operations/kubernetesupgrade.(*Upgrader).upgradeMasterNodes(0xc4208be210, 0x148c101, 0xc4208be210)
	/go/src/github.com/Azure/acs-engine/pkg/operations/kubernetesupgrade/upgrader.go:137 +0xa98
github.com/Azure/acs-engine/pkg/operations/kubernetesupgrade.(*Upgrader).RunUpgrade(0xc4208be210, 0x0, 0x0)
	/go/src/github.com/Azure/acs-engine/pkg/operations/kubernetesupgrade/upgrader.go:54 +0x2f
github.com/Azure/acs-engine/pkg/operations/kubernetesupgrade.(*UpgradeCluster).UpgradeCluster(0xc420a13c60, 0x804f1c7d6b541430, 0x6afe76994af22385, 0xc420b40800, 0x26be, 0x7fff98924da7, 0x1b, 0xc42073b6e0, 0xc420ac9940, 0x8, ...)
	/go/src/github.com/Azure/acs-engine/pkg/operations/kubernetesupgrade/upgradecluster.go:115 +0x660
github.com/Azure/acs-engine/cmd.(*upgradeCmd).run(0xc4200914a0, 0xc42016a900, 0xc420487d00, 0x0, 0x10, 0x0, 0x0)
	/go/src/github.com/Azure/acs-engine/cmd/upgrade.go:225 +0x455
github.com/Azure/acs-engine/cmd.newUpgradeCmd.func1(0xc42016a900, 0xc420487d00, 0x0, 0x10, 0x0, 0x0)
	/go/src/github.com/Azure/acs-engine/cmd/upgrade.go:58 +0x52
github.com/Azure/acs-engine/vendor/github.com/spf13/cobra.(*Command).execute(0xc42016a900, 0xc420487c00, 0x10, 0x10, 0xc42016a900, 0xc420487c00)
	/go/src/github.com/Azure/acs-engine/vendor/github.com/spf13/cobra/command.go:647 +0x3e4
github.com/Azure/acs-engine/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42021b8c0, 0xc42016ab40, 0xc42016a900, 0xc42016a480)
	/go/src/github.com/Azure/acs-engine/vendor/github.com/spf13/cobra/command.go:726 +0x2d4
github.com/Azure/acs-engine/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42021b8c0, 0xc42000e018, 0x13a4b83)
	/go/src/github.com/Azure/acs-engine/vendor/github.com/spf13/cobra/command.go:685 +0x2b
main.main()
	/go/src/github.com/Azure/acs-engine/main.go:12 +0x74
```